src/main/java/seedu/address/ui/ProfileVisitCard.java

```diff
@@ -0,0 +1,86 @@
+package seedu.address.ui;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Region;
+import seedu.address.model.person.VisitReport;
+
+/**
+ * An UI component that displays information of a {@code VisitList}.
+ */
+public class ProfileVisitCard extends UiPart<Region> {
+
+    private static final String FXML = "ProfileVisitCard.fxml";
+
+    public final VisitReport report;
+    public final String date;
+    private String diagnosis;
+    private String medication;
+    private String remarks;
+
+    @FXML
+    private HBox profileVisitCardPane;
+    @FXML
+    private Label profileVisitDate;
+    @FXML
+    private Label profileVisitDiagnosis;
+    @FXML
+    private Label profileVisitMedication;
+    @FXML
+    private Label profileVisitRemarks;
+
+    public ProfileVisitCard(VisitReport report) {
+        super(FXML);
+        this.report = report;
+        this.date = report.date;
+        this.diagnosis = report.getDiagnosis();
+        this.medication = report.getMedication();
+        this.remarks = report.getRemarks();
+
+        // Set date
+        profileVisitDate.setText("Visitation Report on  [" + date + "]");
+
+        // Set Diagnosis
```

Comment:
Maybe we can SLAP this a bit more?

Reply:
Are you referring to line 45-50?

Reply:
I can perhaps refactor this whole setting of labels outside of the constructor, for the next iteration of refactoring.

Reply:
Actually more like setting diagnosis, medication and remarks, extract the method out

Reply:
Yeah thats actually a good idea. I'll do it in the next refactor run.